app/controllers/admin/creator_settings_controller.rb

```diff
@@ -4,34 +4,33 @@ class CreatorSettingsController < Admin::ApplicationController
                         invite_only_mode logo primary_brand_color_hex public].freeze
 
     def new
+      @creator_settings_form = CreatorSettingsForm.new(
+        community_name: ::Settings::Community.community_name,
+        public: ::Settings::UserExperience.public,
+        invite_only_mode: ::Settings::Authentication.invite_only_mode,
+        primary_brand_color_hex: ::Settings::UserExperience.primary_brand_color_hex,
+      )
       @max_file_size = LogoUploader::MAX_FILE_SIZE
       @logo_allowed_types = (LogoUploader::CONTENT_TYPE_ALLOWLIST +
         LogoUploader::EXTENSION_ALLOWLIST.map { |extension| ".#{extension}" }).join(",")
     end
 
     def create
       extra_authorization
-      ActiveRecord::Base.transaction do
-        ::Settings::Community.community_name = settings_params[:community_name]
-        ::Settings::UserExperience.primary_brand_color_hex = settings_params[:primary_brand_color_hex]
-        ::Settings::Authentication.invite_only_mode = settings_params[:invite_only]
-        ::Settings::UserExperience.public = settings_params[:public]
 
-        if settings_params[:logo]
-          logo_uploader = upload_logo(settings_params[:logo])
-          ::Settings::General.original_logo = logo_uploader.url
-          ::Settings::General.resized_logo = logo_uploader.resized_logo.url
-        end
+      @creator_settings_form = CreatorSettingsForm.new(settings_params)
```

Comment:
Instead of handling this here, we call a `CreatorSettingForm` that moves conditional callbacks and validations into a plain old ruby object. This Ruby object handles the updates and validations.

Reply:
@Ridhwana, I ran a quick test by uploading a non-image type, and the page shows the error in the flash message with the form data populated except for the checkboxes.

![image](https://user-images.githubusercontent.com/833231/145458151-4c68e9de-d063-4e05-88d5-44cd4c988f81.png)

This is an example of the Ruby object being returned (I converted it to JSON)

```
{"attributes":{"checked_code_of_conduct":true,"checked_terms_and_conditions":true,"community_name":"ro","invite_only_mode":false,"logo":{"tempfile":"#\u003cFile:0x00007fe1a64a6cc8\u003e","original_filename":"divvy-invoice-2021.pdf","content_type":"application/pdf","headers":"Content-Disposition: form-data; name=\"creator_settings_form[logo]\"; filename=\"divvy-invoice-2021.pdf\"\r\nContent-Type: application/pdf\r\n"},"primary_brand_color_hex":"#3b49df","public":true}}
```

I see the checkbox values in there, but still digging to see why the checkbox values aren't being updated.

Reply:
Ah @nickytonline I don't persist the terms and conditions checkboxes on the user object until all the fields pass. That made the most sense to me and I've seen it done this way before, but if that doesn't make sense to others, we can change it.

Reply:
I thought it was a bug, but I honestly don't know if that's a pattern for accepting a ToS and CoC. I trust your judgement on this one.

Reply:
I just posted in our channel to get product weigh in :) Thanks for bringing it up!

Reply:
@nickytonline  I've gone ahead and persisted the toc and coc as per [the product discussion](https://forem-team.slack.com/archives/C01S4QZUDNC/p1639077971299600). Thanks once again for bringing this up!

![screencapture-localhost-3000-admin-creator-settings-new-2021-12-10-11_37_22](https://user-images.githubusercontent.com/2786819/145552092-358f5cf1-223e-4740-b49e-5f6196576c71.png)